hoomd/error.py

```diff
@@ -0,0 +1,22 @@
+# Copyright (c) 2009-2021 The Regents of the University of Michigan
+# This file is part of the HOOMD-blue project, released under the BSD 3-Clause
+# License.
+
+"""HOOMD Errors."""
```

Comment:
Import `error` in `__init__.py` so that it is available in user namespaces when users `import hoomd`.

Also, add a `module-hoomd-error.rst` that generates the documentation for this module and add it to the proper place in the package index toctree.